### ISSUE TITLE
Ubuntu package 'xroad-base' must be dependent on package 'gpg'

### ISSUE THREAD

**Issue by uvelokk** [nordic-institute/X-Road#1489]
**xroad-base version:** 7.2.0
**OS:** Ubuntu 20.04
**Current dependencies:**
- Pre-Depends: xroad-database-local (= 7.2.0-0.ubuntu20.04) | xroad-database-remote (= 7.2.0-0.ubuntu20.04)
- Depends: debconf (>= 0.5) | debconf-2.0, rsyslog, curl, systemd, xroad-confclient (= 7.2.0-0.ubuntu20.04), xroad-signer (= 7.2.0-0.ubuntu20.04), openjdk-11-jre-headless | java11-runtime-headless | java11-runtime

The file `/usr/share/xroad/scripts/generate_gpg_keypair.sh` from the package 'xroad-base' is run during initial configuration. If the application `gpg` is not installed, the initial configuration ends with error:

```
Caused by: org.niis.xroad.securityserver.restapi.service.ProcessFailedException: Failed to run command '/usr/share/xroad/scripts/generate_gpg_keypair.sh /etc/xroad/gpghome ee-test/xxx/xxxxxxx/xxxxxxx' with output: 
 Generating new GPG keypair for 'ee-test/xxx/xxxxxxx/xxxxxxx'
/usr/share/xroad/scripts/generate_gpg_keypair.sh: line 20: gpg: command not found
	at org.niis.xroad.securityserver.restapi.service.ExternalProcessRunner.executeAndThrowOnFailure(ExternalProcessRunner.java:145)
	at org.niis.xroad.securityserver.restapi.service.InitializationService.generateGPGKeyPair(InitializationService.java:390)
	... 140 common frames omitted
```

Therefore the package 'xroad-base' must be dependent on a package which provides application `gpg`.

**Comment by raits** [comment]
Hello,

Thank you for reporting this to us. I have added it to our JIRA backlog as a bug ticket: https://nordic-institute.atlassian.net/browse/XRDDEV-2230

The aim is to release a fix for this in the near future as part of X-Road 7.2.1.

**Comment by uvelokk** [comment]
UPDATE:
It seems to be dependent on 'gpg-agent' as well:

`2023-01-12T10:47:36+02:00 xxxxx correlation-id: [dc65c0c198a5c712] INFO  [X-Road Proxy Admin REST API] 2023-01-12T10:47:36.451+02:00 - {"event":"Initialize server configuration failed","user":"xxxxxx","reason":"org.niis.xroad.securityserver.restapi.service.ProcessFailedException: Failed to run command '/usr/share/xroad/scripts/generate_gpg_keypair.sh /etc/xroad/gpghome ee-test/xxx/xxxxxxxx/xxxxxxxx' with output: \n Generating new GPG keypair for 'ee-test/xxx/xxxxxxxx/xxxxxxxx'\ngpg: failed to start agent '/usr/bin/gpg-agent': No such file or directory\ngpg: can't connect to the agent: No such file or directory\ngpg: agent_genkey failed: No agent running\ngpg: key generation failed: No agent running","warning":false,"auth":"Session","url":"/api/v1/initialization","data":{"ownerIdentifier":{"memberClass":"xxx","memberCode":"xxxxxxxx","subsystemCode":null,"objectType":"MEMBER","xroadInstance":"ee-test"},"serverCode":"xxxxxxxx"}}`

**Comment by raits** [comment]
Hello,

Sorry for the lack of feedback on this issue. The issue should be solved with the release of version 7.2.1 back in February. If you still have issues regarding GPG, please let us know, and we can investigate further.

I will close this issue for now.
